archunit/src/test/java/com/tngtech/archunit/core/domain/JavaClassTest.java

```diff
@@ -544,28 +552,69 @@ public void direct_dependencies_from_self_by_inheritance_through_generic_interfa
 
         JavaClass javaClass = importClassWithContext(Child.class);
 
+        String genericInterfaceSignature = InterfaceWithTwoTypeParameters.class.getName() + "<"
+                + Comparable.class.getName() + "<" + Child.class.getName() + "<FIRST, SECOND>>, "
+                + Map.class.getName() + "<"
+                + Map.Entry.class.getName() + "<FIRST, " + Map.Entry.class.getName() + "<" + String.class.getName() + ", FIRST>>, "
+                + Map.class.getName() + "<? extends " + BufferedInputStream.class.getName() + "[][], "
+                + Map.class.getName() + "<? extends " + Serializable.class.getName() + ", " + List.class.getName() + "<" + List.class.getName()
+                + "<? extends " + Set.class.getName() + "<? super " + Iterable.class.getName() + "<? super " + Map.class.getName()
+                + "<FIRST, ? extends " + File.class.getName() + ">>>>>>>>>";
         assertThatDependencies(javaClass.getDirectDependenciesFromSelf())
                 .contain(from(Child.class).to(InterfaceWithTwoTypeParameters.class).inLocation(getClass(), 0)
                         .withDescriptionContaining("implements")
 
                         .from(Child.class)
-                        .withExpectedDescriptionTemplate("has generic interface <" + InterfaceWithTwoTypeParameters.class.getName() + "> with type argument depending on <#target>")
+                        .withExpectedDescriptionTemplate("has generic interface <" + genericInterfaceSignature + "> with type argument depending on <#target>")
                         .to(Comparable.class, Map.class, Map.Entry.class, String.class, BufferedInputStream[][].class, Serializable.class, List.class, Set.class, Iterable.class, File.class)
 
                         .from(Child.class).to(BufferedInputStream.class).inLocation(getClass(), 0)
                         .withDescriptionContaining("depends on component type <%s>", BufferedInputStream.class.getName())
                 );
     }
 
+    @Test
+    public void direct_dependencies_from_self_by_field_declarations() {
```

Comment:
Should `SomeClass` have a dependency on `File` in the following example?
```java
        class Outer<T extends File> {
            class SomeClass {
                List<? extends T> field;
            }
        }
```

Reply:
I first thought we could extend `JavaClassDependencies.dependenciesOfTypes` with
```
        } else if (javaType instanceof JavaTypeVariable<?>) {
            result.addAll(dependenciesOfTypes(((JavaTypeVariable<?>) javaType).getBounds()));
```
but that gives me some `StackOverflowError`s... 

I haven't actually looked into them, but dut does this mean that we're missing dependencies from some type's erasure?

Reply:
Hmm, that's an interesting case :thinking: I guess since `SomeClass` can use the bound of `T` the bound must be a dependency...

Reply:
The danger of `StackOverflowError` is not that surprising, since you can define `class Foo<T extends Foo<T>>` :wink:

Reply:
I pondered about it for a while and I think it's actually okay that the inner class does not report the bound of the outer class defined type variable as a dependency per se. Because if the inner class never makes use of that bound, then you could argue that it does not really depend on the bound (you could throw away the bound on the outer type variable and nothing would change for the inner class). However if the inner class would make use of the bound (in your example operate on `File`), then there would be some other dependency on that type present and it would behave reasonably again.

Reply:
If there should be many complaints about this corner case we can still revisit the issue :wink:

Reply:
Sounds totally fair!